### ISSUE TITLE
Bug, dropping FOVs in polygons

### ISSUE THREAD

**Issue by roanvanscheppingen** [dcjones/proseg#26]
Input data has 334 fovs in the table
Each fov has cells identified in Atomx 

The polygon.geojson misses fovs, messing up the FOV numbering.

`> missing_values
[1]  65 176 224 307 315
`

It really becomes evident when I plot the transcripts and the polygons at the same time. After fov 65 you can see a mismatch in them. 
```
cells_by_fov <- split(cell_metadata$cell, cell_metadata$fov)
cells <- unlist(cells_by_fov[69])
filtered_polygons <- subset(polygons, cell %in% cells)
filtered_transcripts <- subset(transcript_metadata, fov == 69)

ggplot() +
  geom_sf(data = filtered_polygons) +
  geom_point(data = filtered_transcripts, aes(x = x, y = y, color = factor(background)), size = 0.001) +
  scale_color_manual(values = c("1" = "#1CE482", "0" = "#1C7EE4")) +
  theme_minimal() +
  labs(title = "Plot of Filtered Polygons in Pixel Space")
```

<img width="551" alt="image" src="https://github.com/user-attachments/assets/255244f6-d28e-4e79-9e4a-ac3c1ef7dfc6">

Input before proseg contains transcripts in all 334 fovs 
Output transcript.metadata contains transcripts in all fovs
It's output polygons that have dropped the polygons and have a new counting order. 

But also cell.metadata that losts some FOVS, so building the Seurat als leads to missing cells. 
Maybe of use, but when starting Proseg, cells identified: 134774, while the atomx says 144636

**Comment by roanvanscheppingen** [comment]
Additionally, an example FOV number 27 is in the cell metadata, but cells are only identified around the edges of the FOV. While in Atomx, there are cells everywhere in the FOV.  See FOV 15 to 28 here, with FOV 27 clearly missing. 
<img width="422" alt="image" src="https://github.com/user-attachments/assets/79fdbcf5-7696-4716-a02d-64aa9e91c81e">

However, I can still plot FOV and transcripts 27, indicating it is in the data, but it still has dropped significant data from the cell (but doesn't lead to a mismatch in transcripts' fov number and polygon fov's number, that only happens after FOV 65. 
<img width="422" alt="image" src="https://github.com/user-attachments/assets/e6a47a18-8c5b-40a5-a600-a9df2cb32065">


I will run proseg again to see if this error persists. It should be noted I have run proseg on this data before, with a different level of segmentation in the Atomx software. I didn't have these problems at that moment. 
* I should still investigate whether the transcripts assigned to a cell in a FOV > 65 are still in the actual cell or somewhere completly different on XY. I'm not sure how downstream analysis is affected currently, except for the loss of cell numbers in the missing FOVs.

**Comment by roanvanscheppingen** [comment]
Reran proseg;

current missing FOVs from cell_metadata =  176, 224, 315 

FOV 69 has now been restored (using code from post 1), since FOV 65 now seems to be in the data. 
<img width="653" alt="image" src="https://github.com/user-attachments/assets/ac27404f-2fd3-4b2e-975e-cbb8331b4179">

A quick plot of all polygons shows more of these mysterious squares with very little cell number or completely absent.
<img width="432" alt="image" src="https://github.com/user-attachments/assets/d73bfc9b-329a-4138-ba4d-2fcb609d2b8b">

```
polygons <- st_read("cell-polygons.geojson")
st_crs(filtered_polygons) <- NA
ggplot() +
  geom_sf(data = polygons) +
  theme_minimal()
```

**Comment by roanvanscheppingen** [comment]
Reran Proseg with less FOVs at the same time, issue persists. 

Atomx FOV 62: 
<img width="480" alt="image" src="https://github.com/user-attachments/assets/8bc3a935-cb55-4182-ac5d-6138efc17038">
Flatfile data, green dots = background and blue dots = in cells, see a long streak of cells in the FOV. 
<img width="635" alt="image" src="https://github.com/user-attachments/assets/f97bde53-b4f0-46c0-8f34-f35799487a1e">
<img width="479" alt="image" src="https://github.com/user-attachments/assets/f5611fd4-5996-4a27-9056-da406ec1dc06">

So far I can conclude that it happens to the FOVs with sparse / worse RNA quality. However, the original input still has some transcripts per cell as you can see. How does proseg model the distribution of the transcripts and decide when it's not enough? 
I have also retraced these analysis to the first times I ran proseg (before atomx resegmentation), and also there the problem persists. Some fovs with lower / unequal distribution of transcripts seem to drop out (almost) completely.

**Comment by dcjones** [comment]
This is very mysterious, I'm not sure what's going on because proseg is initialized with the segmentation from atomx, does not filter cells in way, and doesn't do any fov-specific processing, so I'm at a loss to where it's losing data.

Did you use the `stitch-cosmx.jl` program shipped with proseg or gen a transcript table some other way? Maybe this is what you're plotting already, but did you check that that input file did not drop data from any fovs?

**Comment by dcjones** [comment]
If you're able to share the input file, I'd be happy to investigate this further.

**Comment by dcjones** [comment]
Thanks for your patience with this, I did finally figure it out. Proseg initializes cell with transcripts the prior segmentation assigned to the nuclei. For some reason, no nuclei assignments were made in those fovs, so they were initialized as empty.

I've released version 1.0.6 that adds a couple things that are useful here. First you should be able to run proseg on the transcripts file out of AtoMx with just the `--cosmx` argument, no additional setting of the scale or columns. Second, I've added an option `--use-cell-initialization` to initialize with cell assignments rather than nuclei assignments, which should work around the issue you observed here.
